packages/txservice/src/dispatch.ts

```diff
@@ -675,56 +691,72 @@ export class TransactionDispatch extends ChainRpcProvider {
         transaction: transaction.loggable,
       });
     }
+
     transaction.receipt = receipt;
 
     this.logger.info(`Tx confirmed.`, requestContext, methodContext, {
       chainId: this.chainId,
       receipt: {
-        gasUsed: receipt.gasUsed.toString(),
-        transactionHash: receipt.transactionHash,
-        blockHash: receipt.blockHash,
+        transactionHash: transaction.receipt.transactionHash,
+        confirmations: transaction.receipt.confirmations,
+        blockNumber: transaction.receipt.blockNumber,
       },
+      transactionExecutionTime: Date.now() - transaction.timestamp,
       transaction: transaction.loggable,
     });
     this.callbacks.onConfirm(transaction);
   }
 
   /**
    * Bump the gas price for this tx up by the configured percentage.
+   *
+   * @param transaction - OnchainTransaction object to modify based on bump result.
    */
-  public async bump(transaction: Transaction) {
+  public async bump(transaction: OnchainTransaction) {
     const { requestContext, methodContext } = createLoggingContext(this.bump.name, transaction.context);
-    if (transaction.attempt >= Transaction.MAX_ATTEMPTS) {
-      const error = new MaxAttemptsReached(transaction.attempt, {
-        gasPrice: `${utils.formatUnits(transaction.gas.price, "gwei")} gwei`,
-        transaction: transaction.loggable,
-      });
-      throw error;
-    } else if (transaction.bumps >= transaction.hashes.length) {
+    if (
+      transaction.bumps >= transaction.hashes.length ||
```

Comment:
shouldn't it be `>` rather than `>=` ? dont see why having bumps `==` hashes is a problem (seems like you would want a new hash for each bump)

Reply:
it actually should logically be `===`. The `>` is just my paranoia showing.

we bump *before* sending an additional tx, not after. and we only bump after sending the initial one.

- send initial (hashes length 1)
- bump (hashes length 1, bumps 1)
- send (2, 1)
- bump (2, 2)
...

dont want to bump if we've already bumped for the current hash.